src/builtins/ThresholdPredictors.jl

```diff
@@ -0,0 +1,156 @@
+##
+###  BinaryThresholdPredictor
+##
+"""
+    BinaryThresholdPredictor(wrapped_model=ConstantClassifier(), threshold=0.5)
+    Wraps a `Probabilistic` model(`wrapped_model`) supporting binary classification in a 
+`BinaryThresholdPredictor` model. Training the `BinaryThresholdPredictor` model results in 
+training the underlying `wrapped_model`. 
+`BinaryThresholdPredictor` model predictions are deterministic and depend on the specified 
+probability `threshold`(which lies in [0, 1) ). The positive class is predicted if the 
+probability value exceeds the threshold.
+
```

Comment:
"where the  \"postive\" class is the *second* class returned by `levels(y)`, where `y` is the target. "